core/transaction_pool.go

```diff
@@ -245,16 +278,16 @@ func (self *TxPool) add(tx *types.Transaction) error {
 }
 
 // queueTx will queue an unknown transaction
-func (self *TxPool) queueTx(hash common.Hash, tx *types.Transaction) {
+func (self *TxPool) queueTx(hash common.Hash, tx *types.Transaction, owned bool) {
 	from, _ := tx.From() // already validated
 	if self.queue[from] == nil {
-		self.queue[from] = make(map[common.Hash]*types.Transaction)
+		self.queue[from] = make(map[common.Hash]*poolTx)
 	}
-	self.queue[from][hash] = tx
+	self.queue[from][hash] = &poolTx{tx, owned, time.Now(), time.Time{}, 0}
```

Comment:
Could you write out the field names too here? (i.e. of `poolTx`). With 2 fields it's ok-ish, but 5 is pushing it and can lead to odd issues if we reorganize the order of the fields, add new ones, etc.